sdk/ml/azure-ai-ml/tests/dsl/_util.py

```diff
@@ -15,3 +17,19 @@ def include_private_preview_nodes_in_pipeline():
         yield
     finally:
         PipelineJobSchema._declared_fields["jobs"] = original_jobs
+
+
+def expand_pipeline_nodes(pipeline: Union[PipelineJob, PipelineComponent]):
+    """Expand pipeline nodes to a list of nodes. All sub-graphs will be expanded."""
+    nodes = []
+    for node in pipeline.jobs.values():
+        if isinstance(node, Pipeline):
+            pipeline_component = node.component
+            if not isinstance(pipeline_component, PipelineComponent):
```

Comment:
Will this really happen? In what case?

Reply:
Sometimes the component field will be a string, e.g. when it's from rest